VEDA_OS01/transcripts.py

```diff
@@ -265,3 +296,465 @@ def upload_sjson_to_s3(config, sjson_data):
+    # http://support.3playmedia.com/hc/en-us/articles/227729988-Translations-API-Methods
+    available_services = json.loads(response.text)
+    if not isinstance(available_services, list):
+        # Fail all the pending translation processes associated with this file id.
+        translation_processes.update(status=TranscriptStatus.FAILED)
+
+        raise TranscriptTranslationError(
+            u'[3PlayMedia Callback] Expected list but got: -- {response}.'.format(
+                response=response.text,
+            )
+        )
+
+    for target_language in target_languages:
+
+        try:
+            translation_process = translation_processes.filter(lang_code=target_language).latest()
+        except TranscriptProcessMetadata.DoesNotExist:
+            LOGGER.warning(
+                u'[3PlayMedia Callback] process not found for target language %s -- process id %s',
+                target_language,
+                file_id,
+            )
+            continue
+
+        # 1 - Find a standard service for translation in the target language.
+        translation_service_id = None
+        for service in available_services:
+            service_found = (
+                service['target_language_iso_639_1_code'] == target_language and
+                service['service_level'] == 'standard'
+            )
+            if service_found:
+                translation_service_id = service['id']
+                break
+
+        if translation_service_id is None:
+            # Fail the process
+            translation_process.status = TranscriptStatus.FAILED
+            translation_process.save()
+            LOGGER.error(
+                '[3PlayMedia Callback] No translation service found for target language %s -- process id %s',
+                target_language,
+                file_id,
+            )
+            continue
+
+        # 2 - At this point, we've got our service ready to use. Now, place an order for the translation.
+        response = requests.post(THREE_PLAY_ORDER_TRANSLATION_URL.format(file_id=file_id), data={
+            'apikey': api_key,
+            'api_secret_key': api_secret,
+            'translation_service_id': translation_service_id,
+        })
+
+        if not response.ok:
+            # Fail the process
+            translation_process.status = TranscriptStatus.FAILED
+            translation_process.save()
+            LOGGER.error(
+                '[3PlayMedia Callback] An error occurred during translation, target language=%s, file_id=%s, status=%s',
+                target_language,
+                file_id,
+                response.status_code,
+            )
+            continue
+
+        # Translation Order API returns `success` attribute specifying whether the order has been placed
+        # successfully: http://support.3playmedia.com/hc/en-us/articles/227729988-Translations-API-Methods
+        translation_order = json.loads(response.text)
+        if translation_order.get('success'):
```

Comment:
No we don't need it, I have examined 3Play response it'll never be a list and if it does return a list, an error is fine because this will probably happen when their API is modified.